# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.cpp
# Commit: cc0ca35ee959
# Full Hash: cc0ca35ee9592891e7ca03661d1d1c0a559a13d1
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2022-08-27 17:34:11
# Description:
#   Bug 1787584 - Revert RecvSetCursor to use the correct size comparison.
#   
#   This just converts the comparison to what we had before. aHeight * aStride must be exactly equal to Size().
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155775
# ==============================================================================

diff -r 1a2602dec7ae -r cc0ca35ee959 dom/ipc/BrowserParent.cpp
--- a/dom/ipc/BrowserParent.cpp	Sat Aug 27 09:49:38 2022 +0000
+++ b/dom/ipc/BrowserParent.cpp	Sat Aug 27 10:23:01 2022 +0000
@@ -2334,7 +2334,7 @@
 
   nsCOMPtr<imgIContainer> cursorImage;
   if (aHasCustomCursor) {
-    if (!aCursorData || aHeight * aStride >= aCursorData->Size() ||
+    if (!aCursorData || aHeight * aStride != aCursorData->Size() ||
         aStride < aWidth * gfx::BytesPerPixel(aFormat)) {
       return IPC_FAIL(this, "Invalid custom cursor data");
     }
